### ISSUE TITLE
2.1 has no decorator

### ISSUE THREAD

**Issue by nhumrich** [pyinvoke/invoke#935]
Using the latest fabric and invoke, and fabric is broken due to invoke 2.1.0. The decorator package can't be found or imported by fabric.

```
Traceback (most recent call last):
  File "/home/circleci/.pyenv/versions/3.10.11/lib/python3.10/site-packages/fabric/connection.py", line 7, in <module>
    from invoke.vendor.decorator import decorator
ModuleNotFoundError: No module named 'invoke.vendor.decorator'
```

**Comment by cemkavuklu** [comment]
Experiencing the same issue.
`invoke.vendor.decorator` was deleted in #906 but going through the PR conversation, the reason is not obvious to me.
Some clarification would be much appreciated.

**Comment by bitprophet** [comment]
Yea, I'm not sure @kuwv ever gave a rationale in #906 for why that was removed, and I neglected to follow up on that point before merging.

OTOH this is also arguably bad hygiene on Fabric's part, so I think it's proper to fix it there instead. Fabric will be getting a 3.1 release relatively soon and that feels like the right spot to "add" a "new" dependency. I'll go commit this over there so it's not forgotten.
